pkg/utils/getters/dataGetters.go

```diff
@@ -180,3 +180,18 @@ func RetrieveClusterIDsFromObjectsLabels[T metav1.Object](objectList []T) []stri
 	}
 	return slices.Collect(maps.Keys(clusterIDs))
 }
+
+// RetrieveClusterIDFromObject retrieves the remote cluster ID from the labels of a generic kubernetes object.
+func RetrieveClusterIDFromObject[T metav1.Object](object T) (liqov1beta1.ClusterID, error) {
```

Comment:
I think you can use interfaces here instead of generics, since Object is an interface